FTBFS on kfreebsd

Hello,

Hello fuse3 is still FTBFS on kfreebsd with the following error:
```
[17/56] cc  -o lib/libfuse3.so.3.10.1 lib/libfuse3.so.3.10.1.p/fuse.c.o lib/libfuse3.so.3.10.1.p/fuse_loop.c.o lib/libfuse3.so.3.10.1.p/fuse_loop_mt.c.o lib/libfuse3.so.3.10.1.p/fuse_lowlevel.c.o lib/libfuse3.so.3.10.1.p/fuse_opt.c.o lib/libfuse3.so.3.10.1.p/fuse_signals.c.o lib/libfuse3.so.3.10.1.p/buffer.c.o lib/libfuse3.so.3.10.1.p/cuse_lowlevel.c.o lib/libfuse3.so.3.10.1.p/helper.c.o lib/libfuse3.so.3.10.1.p/modules_subdir.c.o lib/libfuse3.so.3.10.1.p/mount_util.c.o lib/libfuse3.so.3.10.1.p/fuse_log.c.o lib/libfuse3.so.3.10.1.p/mount_bsd.c.o lib/libfuse3.so.3.10.1.p/modules_iconv.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libfuse3.so.3 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now -Wl,--version-script,/<<PKGBUILDDIR>>/lib/fuse_versionscript -pthread -ldl -lrt -Wl,--end-group
FAILED: lib/libfuse3.so.3.10.1
cc  -o lib/libfuse3.so.3.10.1 lib/libfuse3.so.3.10.1.p/fuse.c.o lib/libfuse3.so.3.10.1.p/fuse_loop.c.o lib/libfuse3.so.3.10.1.p/fuse_loop_mt.c.o lib/libfuse3.so.3.10.1.p/fuse_lowlevel.c.o lib/libfuse3.so.3.10.1.p/fuse_opt.c.o lib/libfuse3.so.3.10.1.p/fuse_signals.c.o lib/libfuse3.so.3.10.1.p/buffer.c.o lib/libfuse3.so.3.10.1.p/cuse_lowlevel.c.o lib/libfuse3.so.3.10.1.p/helper.c.o lib/libfuse3.so.3.10.1.p/modules_subdir.c.o lib/libfuse3.so.3.10.1.p/mount_util.c.o lib/libfuse3.so.3.10.1.p/fuse_log.c.o lib/libfuse3.so.3.10.1.p/mount_bsd.c.o lib/libfuse3.so.3.10.1.p/modules_iconv.c.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,libfuse3.so.3 -g -O2 -fdebug-prefix-map=/<<PKGBUILDDIR>>=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-z,relro -Wl,-z,now -Wl,--version-script,/<<PKGBUILDDIR>>/lib/fuse_versionscript -pthread -ldl -lrt -Wl,--end-group
/usr/bin/ld: lib/libfuse3.so.3.10.1.p/mount_util.c.o: in function `fuse_mnt_umount':
./obj-x86_64-kfreebsd-gnu/../lib/mount_util.c:207: undefined reference to `umount2'
collect2: error: ld returned 1 exit status
```

kfreebsd is the FreeBSD kernel and the GNU libC

One of the problem is the fact that [you are testing](https://github.com/libfuse/libfuse/blob/master/lib/mount_util.c#L33) `__FreeBSD__` for the existence of `umount2()` while you should probably test `__FreeBSD_kernel__`

~~The other one looks like meson is using `mount.c` instead of `mount_bsd.c`: https://github.com/libfuse/libfuse/blob/master/lib/meson.build#L7~~

--- a/lib/mount_util.c
+++ b/lib/mount_util.c
@@ -30,7 +30,7 @@
 #include <sys/mount.h>
 #include <sys/param.h>
 
-#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__)
+#if defined(__NetBSD__) || defined(__FreeBSD__) || defined(__DragonFly__) || defined(__FreeBSD_kernel__)
 #define umount2(mnt, flags) unmount(mnt, ((flags) == 2) ? MNT_FORCE : 0)
 #endif
 

